### PR TITLE
Add media query for device size.

### PR DESCRIPTION
**Pull request by sc-deepakgautam**

### CODE REVIEW

File: src/sxastarter/src/components/LayoutContainer.tsx [hunk 1]
```diff
@@ -45,12 +63,19 @@ export const Default = (props: ComponentProps): JSX.Element => {
 
   const id = props.params.RenderingIdentifier;
 
+  const stackColumnAt = props.params.StackColumnAt;
```
Comment (stasmaxymov) [inline src/sxastarter/src/components/LayoutContainer.tsx]: Value of StackColumnAt property should be a breakpoint infix (e.g xs, sm, md). Sending a number value defeats all the benefits of customization and doesn't allow a front-end persona to control/change breakpoints from single place

File: src/sxastarter/src/components/LayoutContainer.tsx [hunk 2]
```diff
@@ -23,6 +26,21 @@ export function parseInlineStyles(styles: string | undefined): Record<string, st
   return styleObject;
 }
 
+export function parseBreakpointsClass(size: number): string {
+  const breakpoints: Record<BreakpointKeys, number> = {
```
Comment (stasmaxymov) [inline src/sxastarter/src/components/LayoutContainer.tsx]: There should be no hardcodes for breakpoints values

File: src/sxastarter/src/components/LayoutContainer.tsx [hunk 3]
```diff
@@ -10,6 +10,9 @@ interface ComponentProps {
   params: ComponentParams;
 }
 
+type BreakpointKeys = 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
```
Comment (stasmaxymov) [inline src/sxastarter/src/components/LayoutContainer.tsx]: Don't introduce any type for breakpoints enumeration as we want customers be able to provide their own names

File: src/sxastarter/src/assets/sass/components/layout-container/_media.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Breakpoints variables
+$breakpoints-sm: 370px;
```
Comment (stasmaxymov) [inline src/sxastarter/src/assets/sass/components/layout-container/_media.scss]: - Use map for storing default values, so customers can add new entries to the map, e.g. botstrap values
$grid-breakpoints: (
  xs: 0,
  sm: 576px,
  md: 768px,
  lg: 992px,
  xl: 1200px,
  xxl: 1400px
);
- prefix breakpoints so it's clear they belong to a layout container
- move variables out to _variables.scss file so it's possible to modify and review them
- declare variables as [default ](https://sass-lang.com/documentation/variables/#default-values) again to simplify modification

Please also test that user can modify and introduce new breakpoints without changes these style files

File: src/sxastarter/src/assets/sass/components/layout-container/_media.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+// Media queries
+@media only screen and (max-width: $breakpoints-sm) {
+  .slc-breakpoints-sm {
+    flex-direction: column;
+    align-items: stretch;
+  }
+}
+
+@media only screen and (min-width:calc(#{$breakpoints-sm + 1})) and (max-width: $breakpoints-md) {
```
Comment (stasmaxymov) [inline src/sxastarter/src/assets/sass/components/layout-container/_media.scss]: Limiting stack with min-width makes no sense. For example when you want to stack at for "Tablet you want it to be stacked for all the screens that less than Tablet  down to 0. You don't want to stack it only for Tablet  but than un-stack for a Phone

File: src/sxastarter/src/assets/sass/components/layout-container/_media.scss [hunk 6]
```diff
@@ -0,0 +1,42 @@
+// Breakpoints variables
+$breakpoints-sm: 370px;
+$breakpoints-md: 768px;
+$breakpoints-lg: 1024px;
+$breakpoints-xl: 1280px;
+$breakpoints-xxl: 1536px;
+
+// Media queries
+@media only screen and (max-width: $breakpoints-sm) {
```
Comment (stasmaxymov) [inline src/sxastarter/src/assets/sass/components/layout-container/_media.scss]: currently you hardcoded @media queries for each breakpoint, making it impossible to introduce new ones by customer. You should iterate over a breakpoints map

**Review by stasmaxymov — CHANGES_REQUESTED** [review]

File: src/sxastarter/src/components/LayoutContainer.tsx (hunk 1 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/components/LayoutContainer.tsx]: Updated!

File: src/sxastarter/src/components/LayoutContainer.tsx (hunk 2 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/components/LayoutContainer.tsx]: Updated!

File: src/sxastarter/src/components/LayoutContainer.tsx (hunk 3 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/components/LayoutContainer.tsx]: Removed!

File: src/sxastarter/src/assets/sass/components/layout-container/_media.scss (hunk 4 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/assets/sass/components/layout-container/_media.scss]: Updated!

File: src/sxastarter/src/assets/sass/components/layout-container/_media.scss (hunk 6 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/assets/sass/components/layout-container/_media.scss]: Removed!

File: src/sxastarter/src/assets/sass/components/layout-container/index.scss [hunk 7]
```diff
@@ -130,3 +132,15 @@
...
   }
 }
+
+@each $breakpoint, $value in $slc-grid-breakpoints {
+  $infix: breakpoint-infix($breakpoint, $slc-grid-breakpoints);
+
+  @if ($infix != '') {
+    @media only screen and (max-width: $value) {
+      .slc#{$infix}-stack {
+        flex-direction: column;
```
Comment (stasmaxymov) [inline src/sxastarter/src/assets/sass/components/layout-container/index.scss:142]: builder.io in their stack at implementation also use ` align-items: stretch` property. Do we need it in our solution?

File: src/sxastarter/src/components/LayoutContainer.tsx [hunk 8]
```diff
@@ -45,12 +47,23 @@ export const Default = (props: ComponentProps): JSX.Element => {
 
   const id = props.params.RenderingIdentifier;
 
+  let columnStackClass = '';
+  const stackAtBreakpoint = props.params.StackColumnAt;
```
Comment (stasmaxymov) [inline src/sxastarter/src/components/LayoutContainer.tsx:51]: Rendering parameters might be empty or underfined sometimes, let's also cover those

**Review by stasmaxymov — APPROVED** [review]

File: src/sxastarter/src/components/LayoutContainer.tsx (hunk 8 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/components/LayoutContainer.tsx:51]: Updated!

File: src/sxastarter/src/assets/sass/components/layout-container/index.scss (hunk 7 above)
Comment (sc-deepakgautam) [reply src/sxastarter/src/assets/sass/components/layout-container/index.scss:142]: Looks like its good to set this property. I have added it now.

**Pull request merged**